tests/add_ekubo_pools.cairo

```diff
@@ -0,0 +1,319 @@
+use amm_governance::traits::{IERC20Dispatcher, IERC20DispatcherTrait};
+use amm_governance::types::Option_;
+use core::traits::{Into, TryInto};
+
+use cubit::f128::types::{Fixed, FixedTrait};
+use konoha::upgrades::IUpgradesDispatcher;
+use konoha::upgrades::IUpgradesDispatcherTrait;
+use snforge_std::{
+    CheatSpan, CheatTarget, ContractClassTrait, ContractClass, start_prank, prank, start_warp,
+    declare
+};
+use starknet::ClassHash;
+use starknet::SyscallResult;
+use starknet::SyscallResultTrait;
+use starknet::syscalls::deploy_syscall;
+
+
+use starknet::{ContractAddress, get_block_timestamp};
+
+fn get_voter_addresses() -> @Span<felt252> {
+    let arr = array![
+        0x0011d341c6e841426448ff39aa443a6dbb428914e05ba2259463c18308b86233,
+        0x0583a9d956d65628f806386ab5b12dccd74236a3c6b930ded9cf3c54efc722a1,
+        0x03d1525605db970fa1724693404f5f64cba8af82ec4aab514e6ebd3dec4838ad,
+        0x00d79a15d84f5820310db21f953a0fae92c95e25d93cb983cc0c27fc4c52273c,
+        0x0428c240649b76353644faF011B0d212e167f148fdd7479008Aa44eEaC782BfC,
+        0x06717eaf502baac2b6b2c6ee3ac39b34a52e726a73905ed586e757158270a0af,
+    ];
+    @arr.span()
+}
+
+
+#[test]
+#[fork("MAINNET", block_number: 818610)]
+fn test_add_ekubo_options() {
+    let gov_addr = 0x001405ab78ab6ec90fba09e6116f373cda53b0ba557789a4578d8c1ec374ba0f
+        .try_into()
+        .unwrap();
+
+    // Transfer ownership to governance
+    let amm_owner_address: ContractAddress =
+        0x74fd7da23e21f0f0479adb435221b23f57ca4c32a0c68aad9409a41c27f3067
+        .try_into()
+        .unwrap();
+    let amm_address: ContractAddress =
+        0x047472e6755afc57ada9550b6a3ac93129cc4b5f98f51c73e0644d129fd208d9
+        .try_into()
+        .unwrap();
+    let amm = IAMMDispatcher { contract_address: amm_address };
+
+    prank(CheatTarget::One(amm_address), amm_owner_address, CheatSpan::TargetCalls(1));
+    // start_prank(CheatTarget::One(amm_address), amm_owner_address);
+    amm.transfer_ownership(gov_addr);
+
+    // Declare the new arbitrary proposal
+    let arb_prop_class = declare("ArbitraryProposalAddEkuboPools");
+    let props = IProposalsDispatcher { contract_address: gov_addr };
+
+    let user1: ContractAddress =
+        0x0011d341c6e841426448ff39aa443a6dbb428914e05ba2259463c18308b86233 // team m 1
+        .try_into()
+        .unwrap();
+
+    start_prank(CheatTarget::One(gov_addr), user1);
+
+    let prop_id = props.submit_proposal(arb_prop_class.unwrap().class_hash.into(), 6);
+
+    let mut voter_addresses = *get_voter_addresses();
+    // vote yay with all users
+    loop {
+        match voter_addresses.pop_front() {
+            Option::Some(address) => {
+                let current_voter: ContractAddress = (*address).try_into().unwrap();
+                prank(CheatTarget::One(gov_addr), current_voter, CheatSpan::TargetCalls(1));
+                props.vote(prop_id, 1);
+            },
+            Option::None(()) => { break (); }
+        }
+    };
```

Comment:
did you know the famed for loop has come to our land?

https://community.starknet.io/t/cairo-v2-7-0-is-coming/114362

(not worth changing, just wanted to mention)